src/vellum/workflows/nodes/bases/base.py

```diff
@@ -457,21 +457,6 @@ def __init__(
             resolved_value = resolve_value(descriptor.instance, self.state, path=descriptor.name, memo=inputs)
             setattr(self, descriptor.name, resolved_value)
 
-        # Resolve descriptors set as defaults to the outputs class
-        def _outputs_post_init(outputs_self: "BaseNode.Outputs", **kwargs: Any) -> None:
-            for node_output_descriptor in self.Outputs:
-                if node_output_descriptor.name in kwargs:
-                    continue
-
-                if isinstance(node_output_descriptor.instance, BaseDescriptor):
-                    setattr(
-                        outputs_self,
-                        node_output_descriptor.name,
-                        node_output_descriptor.instance.resolve(self.state),
-                    )
-
-        setattr(self.Outputs, "_outputs_post_init", _outputs_post_init)
```

Comment:
this was the root of the bug.

`_outputs_post_init` was referencing instances in its method (`self.state`), even though it was assigned to a class (`self.Outputs`). So parallel executions would keep overwriting `self.Outputs` in a non-thread-safe way.

Moral of the story here is do not do instance operations in class methods